<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1316</width>
    <height>708</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Traffic Network Optimization</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tabWidget_2">
       <attribute name="title">
        <string>Graph</string>
       </attribute>
       <widget class="QWidget" name="layoutGraph" native="true">
        <property name="geometry">
         <rect>
          <x>340</x>
          <y>10</y>
          <width>941</width>
          <height>551</height>
         </rect>
        </property>
       </widget>
       <widget class="QFrame" name="frame">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>-10</y>
          <width>341</width>
          <height>601</height>
         </rect>
        </property>
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <widget class="QComboBox" name="comboAlgorithm">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>60</y>
           <width>91</width>
           <height>19</height>
          </rect>
         </property>
         <item>
          <property name="text">
           <string>GA</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>BAT</string>
          </property>
         </item>
        </widget>
        <widget class="QPushButton" name="btnLoadCSV">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>120</y>
           <width>91</width>
           <height>23</height>
          </rect>
         </property>
         <property name="text">
          <string>Chọn file mạng</string>
         </property>
        </widget>
        <widget class="QPushButton" name="btnRun">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>180</y>
           <width>91</width>
           <height>23</height>
          </rect>
         </property>
         <property name="text">
          <string>Chạy tối ưu</string>
         </property>
        </widget>
        <widget class="QLabel" name="lblFileName">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>150</y>
           <width>91</width>
           <height>24</height>
          </rect>
         </property>
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;Chưa chọn file&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
        <widget class="QLabel" name="labelAlgorithm">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>30</y>
           <width>91</width>
           <height>24</height>
          </rect>
         </property>
         <property name="text">
          <string>Chọn thuật toán:</string>
         </property>
        </widget>
        <widget class="QLabel" name="lblStatus">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>470</y>
           <width>321</width>
           <height>41</height>
          </rect>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
        <widget class="QLineEdit" name="txtEK">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>90</y>
           <width>91</width>
           <height>19</height>
          </rect>
         </property>
         <property name="text">
          <string>EK</string>
         </property>
        </widget>
        <widget class="QStackedWidget" name="stackProperties">
         <property name="geometry">
          <rect>
           <x>100</x>
           <y>110</y>
           <width>231</width>
           <height>341</height>
          </rect>
         </property>
         <property name="currentIndex">
          <number>1</number>
         </property>
         <widget class="QWidget" name="page">
          <widget class="QGroupBox" name="grppsoi">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>0</y>
             <width>221</width>
             <height>331</height>
            </rect>
           </property>
           <property name="title">
            <string/>
           </property>
           <widget class="QSpinBox" name="spinBat">
            <property name="geometry">
             <rect>
              <x>130</x>
              <y>10</y>
              <width>61</width>
              <height>20</height>
             </rect>
            </property>
           </widget>
           <widget class="QLabel" name="label_2">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>60</y>
              <width>101</width>
              <height>34</height>
             </rect>
            </property>
            <property name="text">
             <string>Tần số nhỏ nhất</string>
            </property>
           </widget>
           <widget class="QLabel" name="label_6">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>150</y>
              <width>101</width>
              <height>34</height>
             </rect>
            </property>
            <property name="text">
             <string>Loudness Min</string>
            </property>
           </widget>
           <widget class="QDoubleSpinBox" name="spinPratemax">
            <property name="geometry">
             <rect>
              <x>130</x>
              <y>220</y>
              <width>71</width>
              <height>20</height>
             </rect>
            </property>
           </widget>
           <widget class="QLabel" name="label_3">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>30</y>
              <width>51</width>
              <height>34</height>
             </rect>
            </property>
            <property name="text">
             <string>Loops</string>
            </property>
           </widget>
           <widget class="QLabel" name="label">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>0</y>
              <width>51</width>
              <height>33</height>
             </rect>
            </property>
            <property name="text">
             <string>Bat</string>
            </property>
           </widget>
           <widget class="QDoubleSpinBox" name="spinfmax">
            <property name="geometry">
             <rect>
              <x>130</x>
              <y>100</y>
              <width>71</width>
              <height>20</height>
             </rect>
            </property>
           </widget>
           <widget class="QSpinBox" name="spinLoops">
            <property name="geometry">
             <rect>
              <x>130</x>
              <y>40</y>
              <width>71</width>
              <height>20</height>
             </rect>
            </property>
            <property name="maximum">
             <number>999999</number>
            </property>
           </widget>
           <widget class="QDoubleSpinBox" name="spinLoudmax">
            <property name="geometry">
             <rect>
              <x>130</x>
              <y>160</y>
              <width>71</width>
              <height>19</height>
             </rect>
            </property>
           </widget>
           <widget class="QLabel" name="label_4">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>90</y>
              <width>101</width>
              <height>34</height>
             </rect>
            </property>
            <property name="text">
             <string>Tần số lớn nhất</string>
            </property>
           </widget>
           <widget class="QDoubleSpinBox" name="spinfmin">
            <property name="geometry">
             <rect>
              <x>130</x>
              <y>70</y>
              <width>71</width>
              <height>19</height>
             </rect>
            </property>
           </widget>
           <widget class="QDoubleSpinBox" name="spinLoudmin">
            <property name="geometry">
             <rect>
              <x>130</x>
              <y>130</y>
              <width>71</width>
              <height>20</height>
             </rect>
            </property>
           </widget>
           <widget class="QLabel" name="label_7">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>180</y>
              <width>91</width>
              <height>34</height>
             </rect>
            </property>
            <property name="text">
             <string>Pulse rate Min</string>
            </property>
           </widget>
           <widget class="QLabel" name="label_5">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>120</y>
              <width>101</width>
              <height>33</height>
             </rect>
            </property>
            <property name="text">
             <string>Loudness Max</string>
            </property>
           </widget>
           <widget class="QLabel" name="label_10">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>210</y>
              <width>91</width>
              <height>34</height>
             </rect>
            </property>
            <property name="text">
             <string>Pulse rate Max</string>
            </property>
           </widget>
           <widget class="QDoubleSpinBox" name="spinPratemin">
            <property name="geometry">
             <rect>
              <x>130</x>
              <y>190</y>
              <width>71</width>
              <height>20</height>
             </rect>
            </property>
           </widget>
           <widget class="QLabel" name="label_11">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>240</y>
              <width>71</width>
              <height>34</height>
             </rect>
            </property>
            <property name="text">
             <string>Alpha</string>
            </property>
           </widget>
           <widget class="QDoubleSpinBox" name="spinAlpha">
            <property name="geometry">
             <rect>
              <x>130</x>
              <y>250</y>
              <width>71</width>
              <height>20</height>
             </rect>
            </property>
           </widget>
           <widget class="QLabel" name="label_12">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>280</y>
              <width>71</width>
              <height>21</height>
             </rect>
            </property>
            <property name="text">
             <string>Gamma</string>
            </property>
           </widget>
           <widget class="QDoubleSpinBox" name="spinGamma">
            <property name="geometry">
             <rect>
              <x>130</x>
              <y>280</y>
              <width>71</width>
              <height>20</height>
             </rect>
            </property>
           </widget>
          </widget>
         </widget>
         <widget class="QWidget" name="page_2">
          <widget class="QGroupBox" name="groupBox">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>10</y>
             <width>211</width>
             <height>301</height>
            </rect>
           </property>
           <property name="title">
            <string/>
           </property>
           <widget class="QSpinBox" name="spinpop_size">
            <property name="geometry">
             <rect>
              <x>130</x>
              <y>10</y>
              <width>61</width>
              <height>20</height>
             </rect>
            </property>
           </widget>
           <widget class="QSpinBox" name="spingen_max">
            <property name="geometry">
             <rect>
              <x>130</x>
              <y>40</y>
              <width>71</width>
              <height>20</height>
             </rect>
            </property>
            <property name="maximum">
             <number>999999</number>
            </property>
           </widget>
           <widget class="QDoubleSpinBox" name="spinmuti_rate">
            <property name="geometry">
             <rect>
              <x>130</x>
              <y>70</y>
              <width>71</width>
              <height>19</height>
             </rect>
            </property>
           </widget>
           <widget class="QDoubleSpinBox" name="spincross_rate">
            <property name="geometry">
             <rect>
              <x>130</x>
              <y>100</y>
              <width>71</width>
              <height>19</height>
             </rect>
            </property>
           </widget>
           <widget class="QLabel" name="label_16">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>10</y>
              <width>111</width>
              <height>20</height>
             </rect>
            </property>
            <property name="text">
             <string>Kích thước quần thể</string>
            </property>
           </widget>
           <widget class="QLabel" name="label_17">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>40</y>
              <width>101</width>
              <height>20</height>
             </rect>
            </property>
            <property name="text">
             <string>Số thế hệ tối đa</string>
            </property>
           </widget>
           <widget class="QLabel" name="label_18">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>70</y>
              <width>81</width>
              <height>20</height>
             </rect>
            </property>
            <property name="text">
             <string>Tỉ lệ đột biến</string>
            </property>
           </widget>
           <widget class="QLabel" name="label_19">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>100</y>
              <width>81</width>
              <height>20</height>
             </rect>
            </property>
            <property name="text">
             <string>Tỉ lệ lai ghép</string>
            </property>
           </widget>
          </widget>
         </widget>
        </widget>
        <widget class="QLabel" name="label_20">
         <property name="geometry">
          <rect>
           <x>120</x>
           <y>60</y>
           <width>81</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Điểm bắt đầu</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="txtSource">
         <property name="geometry">
          <rect>
           <x>240</x>
           <y>60</y>
           <width>61</width>
           <height>20</height>
          </rect>
         </property>
        </widget>
        <widget class="QLabel" name="label_21">
         <property name="geometry">
          <rect>
           <x>120</x>
           <y>90</y>
           <width>81</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Điểm kết thúc</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="txtSink">
         <property name="geometry">
          <rect>
           <x>240</x>
           <y>90</y>
           <width>61</width>
           <height>20</height>
          </rect>
         </property>
        </widget>
        <widget class="QLabel" name="label_13">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>440</y>
           <width>321</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:11pt;&quot;&gt;Trạng thái hiện tại&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Compare results</string>
       </attribute>
       <widget class="QTextEdit" name="textEdit">
        <property name="geometry">
         <rect>
          <x>1070</x>
          <y>20</y>
          <width>221</width>
          <height>491</height>
         </rect>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
       <widget class="QFrame" name="frame_2">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>10</y>
          <width>1041</width>
          <height>611</height>
         </rect>
        </property>
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QWidget" name="layoutConvergence" native="true"/>
         </item>
         <item>
          <widget class="QWidget" name="layoutEsmond" native="true"/>
         </item>
        </layout>
       </widget>
       <widget class="QLabel" name="label_8">
        <property name="geometry">
         <rect>
          <x>1080</x>
          <y>0</y>
          <width>211</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;&lt;span style=&quot; font-size:11pt;&quot;&gt;Verbose&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
       <widget class="QLabel" name="txtMaxFlowEK">
        <property name="geometry">
         <rect>
          <x>1070</x>
          <y>550</y>
          <width>221</width>
          <height>31</height>
         </rect>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
       <widget class="QLabel" name="label_9">
        <property name="geometry">
         <rect>
          <x>1070</x>
          <y>520</y>
          <width>221</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p align=&quot;center&quot;&gt;&lt;span style=&quot; font-size:11pt;&quot;&gt;Max flow của đồ thị&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1316</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuRun">
    <property name="title">
     <string>Run</string>
    </property>
    <addaction name="actionOptimize"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuRun"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionOpen">
   <property name="text">
    <string>Open Network</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionOptimize">
   <property name="text">
    <string>Optimize</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>spinBat</tabstop>
  <tabstop>spinLoops</tabstop>
  <tabstop>spinfmin</tabstop>
  <tabstop>spinfmax</tabstop>
  <tabstop>spinLoudmin</tabstop>
  <tabstop>spinLoudmax</tabstop>
  <tabstop>spinPratemax</tabstop>
  <tabstop>comboAlgorithm</tabstop>
  <tabstop>txtEK</tabstop>
  <tabstop>btnLoadCSV</tabstop>
  <tabstop>btnRun</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
